File: sdk/python/exemplos/01_login.py
"""Exemplo 1/10 — login por usuário e senha, primeiro token de serviço.

`Plataforma.entrar()` faz `POST /api/login` e, com a sessão resultante, `POST /api/tokens`
(que só aceita sessão de cookie — nunca outro token). O SDK devolve um `Plataforma` já pronto
para uso por token dali em diante; é o padrão para o resto dos exemplos."""

from __future__ import annotations

from plat import Plataforma


def main(url: str, inquilino: str, login: str, senha: str) -> dict:
    p = Plataforma.entrar(url, inquilino, login, senha, nome_token="sdk-python-exemplo-01")
    try:
        eu = p.eu()
        assert eu["login"] == login
        assert p.token.startswith("plat_")
        print(f"login ok: {eu['login']} (inquilino {eu['inquilino']['nome']!r}, perfil {eu['perfil']!r})")
        return eu
    finally:
        p.tokens.revogar(p.token_id)
        p.fechar()


if __name__ == "__main__":
    from _ambiente import do_ambiente

    main(*do_ambiente())

File: web/sdk/exemplos/01_login.js
/* Exemplo 1/10 — login por usuário e senha, primeiro token de serviço; depois `eu()` por Bearer. */
import { Plataforma, preparar } from './_comum.js';

preparar('01_login', async ({ url, inquilino, login, senha, escrever }) => {
  const p = await Plataforma.entrar(url, inquilino, login, senha, { nomeToken: 'sdk-js-exemplo-01' });
  try {
    const eu = await p.eu();
    if (eu.login !== login) throw new Error(`eu.login ${eu.login} != ${login}`);
    if (!p.token.startsWith('plat_')) throw new Error('token sem prefixo plat_');
    escrever(`login ok: ${eu.login} (inquilino ${eu.inquilino.nome}, perfil ${eu.perfil})`);
    escrever(`token de serviço criado (id ${p.tokenId}); ${p}`); // toString nunca imprime o token
    return eu;
  } finally {
    await p.sair();
    escrever('token revogado; sessão encerrada');
  }
});
